django/contrib/contenttypes/fields.py

```diff
@@ -237,7 +237,10 @@ def __get__(self, instance, cls=None):
                     pk = rel_obj._meta.pk
                     # If the primary key is a remote field, use the referenced
                     # field's to_python().
-                    pk_to_python = pk.target_field.to_python if pk.remote_field else pk.to_python
+                    to_python_field = pk
+                    while to_python_field.remote_field:
```

Comment:
I have been trying to think of cases where this could lead to an infinite loop and the following simplified scenario , while unlikely, could cause issue.

```python
class A(Model):
    b = models.ForeignKey('B', CASCADE, primary_key=True)

class B(Model):
    a = models.ForeignKey('A', CASCADE, primary_key=True)
```

Since this is going to be backported to a LTS I'd favor we rely on a memo to prevent this from happening just to be extra safe.

```python
to_python_field = pk
seen = set(to_python_field)
while to_python_field.remote_field:
    to_python_field = to_python_field.target_field
    if to_python_field in seen:
        break
    seen.add(to_python_field)

Reply:
Good point; will do.

Reply:
Would you want a unittest for that?

Reply:
@morganwahl yes please!